hazelcast/serialization/input.py

```diff
@@ -111,7 +111,7 @@ def read_byte_array(self):
         result = bytearray(length)
         if length > 0:
             self.read_into(result, 0, length)
-        return [x for x in result]
+        return result
```

Comment:
@mdumandag do you reckon this is a breaking change? I think the method should expect a bytearray input and produce bytearray output. if thats not the case, its the responsibility of the caller to address it, similar to what I did with the tests. We can add this in the release notes, but in my opinion its worth the pain, because it give good performance.

Reply:
I agree with you, this method should return a `bytearray` but doing it now is a breaking change since this method is in the public API. If we are going to release these fixes as 3.12.1 I think we should just return `list(result)`. @asimarslan what do you think ?

Reply:
Thinking about it a bit more, how about if we keep the public API unchanged with a default arg? something like `as_list=True`. Updated the commit, have a look

Reply:
@tkountis @mdumandag Can you validate the change with our backward compatibility tests? if the tests fail we can accept it. We never break backward compatibility for any enhancement for default behavior. We can only add a system property enabled feature that has breaking changes.

Reply:
looks good to me

Reply:
@tkountis I think we'll remove that parameter in 4.0 so putting a global config and keeping the public API clear make more sense to me. Maybe a new clientProperties config

Reply:
Done, @asimarslan @mdumandag please have another look